src/js/tech/html5.js

```diff
@@ -203,6 +207,45 @@ class Html5 extends Tech {
     });
   }
 
+  /**
+   * Handle IOS Headphone disconnection during playback
+   *
+   * @private
+  */
+  handleIOSHeadphonesDisconnection_() {
+    // Fudge factor to account for TimeRanges rounding
+    const TIME_FUDGE_FACTOR = 1 / 30;
+
+    // Comparisons between time values such as current time and the end of the buffered range
+    // can be misleading because of precision differences or when the current media has poorly
+    // aligned audio and video, which can cause values to be slightly off from what you would
+    // expect. This value is what we consider to be safe to use in such comparisons to account
+    // for these scenarios.
+    const SAFE_TIME_DELTA = TIME_FUDGE_FACTOR * 3;
+
+    // If iOS check if we have a real stalled or supend event or
+    // we got stalled/suspend due headphones where disconnected during playback
+    this.on(['stalled', 'suspend'], (e) => {
+      const buffered = this.buffered();
+
+      if (buffered.length > 0) {
+        let extraBuffer = false;
+
+        // Establish if we have an extra buffer in the current time range playing.
+        for (let i = 0; i < buffered.length; i++) {
+          if (buffered.start(i) <= this.currentTime() <= buffered.end(i) + SAFE_TIME_DELTA) {
```

Comment:
1. I think what we want is to check if there is anything in the forward buffer. It seems that if the `currentTime` is equal to the `buffered.end`, then we don't have anything in the forward buffer. Maybe just `buffered.start <= currentTime < buffered.end`?
1. Should we adjust the `buffered.start` using `SAFE_TIME_DELTA` as well? Maybe not.
1. We might as well `break` out of the loop when we set `extraBuffer = true`.